sdcm/cluster_oci.py

```
@@ -89,7 +96,160 @@ def wait_for_cloud_init(self):
+        """Try to set scylla_network_configuration from OCI API data (before SSH is available).
+
+        If IPv6 addresses aren't available yet from the API, this will log a warning
+        and leave scylla_network_configuration as None for post-SSH discovery.
+        """
+        try:
+            network_config = ScyllaNetworkConfiguration(
+                network_interfaces=self.network_interfaces,
+                scylla_network_config=scylla_network_config,
+            )
+            # Validate that test_communication resolves (catches empty ipv6_public_addresses)
+            network_config.test_communication
```

Ruff (0.16.0)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)